content/PGP/YubiKey 5.2.3 Enhancements to OpenPGP 3.4.adoc

```diff
@@ -0,0 +1,168 @@
+
+=== General Enhancements
+The OpenPGP Smart card 3.4 specifications include a number of enhancements designed to improve general functionality and speed. These include:
+
+* Expanded Length Responses
+* Algorithm Information & Attributes
+* Key Information
+* Get Random Challenge
+
+=== Expanded Length Responses
+
+The YubiKey firmware 5.2.3 and up can utilize longer responses to queries from OpenPGP, allowing more data to be sent per interaction and reduce the overall time for operations, especially in environments where the USB communication latency is the largest bottleneck. Several data objects (DOs) with variable length have had their maximum response length increased with the max length now listed in the Extended Capabilities DO. The YubiKey has had the max response length increased from 255 to over 3000 bytes. Section 4.4.1 in the OpenPGP Smart Card 3.4 spec lists the DOs which have been affected by this enhancement.
+
+=== Algorithm Information & Attributes
+When generating or importing new keys with the new expanded algorithm set, it is important for the OpenPGP Smart Card to correctly identify which algorithms are supported. The Algorithm Information and Attributes DOs provide that information to a host system. 
+
+The Algorithm Information DO will list all of the supported algorithms and key sizes for an OpenPGP Smart card. This allows the OpenPGP host to correctly present options to users for everything supported on the Authentication, Signature and Decipher keys. Applications can read this DO without having to make changes to the algorithms on the associated key. The Algorithm Information is listed in section 4.4.3.11 of the https://www.google.com/url?q=https://gnupg.org/ftp/specs/OpenPGP-smart-card-application-3.4.pdf&sa=D&ust=1566260477106000[OpenPGP Smart Card 3.4] spec.
+
+Each key on an OpenPGP card can also have the allowed algorithms and keys sizes set to a specific subset of all available options. The allowed options are set in the Algorithm Attributes DO. The attributes can be changed independent for each key, so it is possible, for example, to use different key length for signing and decrypting. If the attributes of an existing key are changed and no longer match with the stored key, the card will prevent it from being used. The Algorithm Attributes is listed in section 4.4.3.9 of the https://www.google.com/url?q=https://gnupg.org/ftp/specs/OpenPGP-smart-card-application-3.4.pdf&sa=D&ust=1566260477106000[OpenPGP Smart Card 3.4] spec.

```

Comment:
I'm not sure about this section.  You could possibly just drop it.  It's not a new feature, at least, but maybe it is still worth pointing out that each key is permitted to have different attributes. 

"the card will prevent it from being used" isn't quite correct; the key is completely wiped if its attributes are changed.  It's worth changing that if keeping this section, as the user should realize it's a destructive action.